[mercadopago-sdk-js] Improve params and returns from constructors and submits

francineemilia opened the pull request:
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:
If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes:
For changes in constructors, they were made based on this doc: https://www.mercadopago.com/developers/en/docs/checkout-bricks/common-initialization
For changes related to submits, they were made based on these docs:
https://www.mercadopago.com/developers/en/docs/checkout-bricks/status-screen-brick/default-rendering
https://www.mercadopago.com/developers/en/docs/checkout-bricks/wallet-brick/default-rendering
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

francineemilia committed:
chore: turns options as optional parameter to init mercadopago

markscamilleri committed:
🤖 Merge PR #65607 [@types/aws-lambda] Fixing SNSMessage field names to match the AWS developer guide by @markscamilleri

* Updated field names to match AWS' developer guide

* Removed accidental auto-formatting

* Ran prettier as per linting instructions

* Fixed tslint:disable-next-line due to prettier adding lines

* Added optional marker for Subject to solve #54379

* Revert "Added optional marker for Subject to solve #54379"

This reverts commit 539abca3902aefdd07e1132eaa45ec429568f84a.

francineemilia committed:
fix: turn style param in brick constructor optional

francineemilia committed:
fix: allows return from onSubmit to be string

thieman committed:
🤖 Merge PR #65623 [memory-cache] Fix del sig, return boolean instead of void by @thieman

yoursunny committed:
🤖 Merge PR #65785 default-gateway: upgrade to v7.2 by @yoursunny

adamcrume committed:
🤖 Merge PR #65683 Update cytoscape core style API and make collections iterable by @adamcrume

mstoykov committed:
🤖 Merge PR #65765 Run prettier on the k6 types as suggested in the common mistakes by @mstoykov

PeterShershov committed:
🤖 Merge PR #65777 [Mixpanel] Update types of register methods by @PeterShershov

* Update types of register methods
- added test for the days and options object
- added tests for unregister method signature in mixpanel-browser

* formatting

* make persistent optional in the mixpanel package

aj-richardson committed:
🤖 Merge PR #65790 Jasmine - Make createSpyObj work when toString is overridden by @aj-richardson

* Make createSpyObj work when toString is overridden

If a class overrides toString (or any Object prototype method), then TS
would previously throw an error:
"Types of property 'toString' are incompatible.
Type '() => string' is not assignable to type 'string'."

* Add a test verifying object spies didn't break.

gabritto committed:
🤖 Merge PR #65796 [jsonld] Use different name for local type and variable by @gabritto

gabritto committed:
🤖 Merge PR #65795 [relay-runtime] Remove unneeded interface by @gabritto

JoshuaKGoldberg committed:
🤖 Merge PR #65786 [npm-whoami] Add types by @JoshuaKGoldberg

* [npm-whoami] Add types

* Update types/npm-whoami/npm-whoami-tests.ts

* Update types/npm-whoami/index.d.ts

Co-authored-by: Piotr Błażejewicz (Peter Blazejewicz) <peterblazejewicz@users.noreply.github.com>

* Whoami as prefix

---------

Co-authored-by: Piotr Błażejewicz (Peter Blazejewicz) <peterblazejewicz@users.noreply.github.com>

jreyesr committed:
🤖 Merge PR #65614 Fix convert() function signatures on Carbone (carbone.io) types by @jreyesr

* Fix convert() function signatures

Removes convertTo parameter on convert() functions
API change happened on Carbone commit bb858ce, Dec 6, 2020, released on tag 3.1.0

* Update carbone-tests.ts

* Update index.d.ts

* Add untyped functions discovered by Danger

* Fix formatter issues

* Make prefixLength optional

leandrodotec committed:
🤖 Merge PR #65788 Add types definitions for xk6-sql by @leandrodotec

* Add types definitions for xk6-sql

Adding definitions for k6/x/sql plugin: https://github.com/grafana/xk6-sql

* remove file

* adding "-browser" suffix for non npm packages.

* more guess work

* fixing identations

* Update types/xk6-sql-browser/index.d.ts

Co-authored-by: Piotr Błażejewicz (Peter Blazejewicz) <peterblazejewicz@users.noreply.github.com>

* Fixing "npm-naming": false and re-running prettier

---------

Co-authored-by: Leandro Gomes <leandro.gomes@bpsoftware.net>
Co-authored-by: Piotr Błażejewicz (Peter Blazejewicz) <peterblazejewicz@users.noreply.github.com>

typescript-bot committed:
🤖 Update CODEOWNERS

mstoykov committed:
🤖 Merge PR #65764 k6 v0.45.0 updates by @mstoykov

* Add metadata JS API and maintainance around it

* k6: Add k6/experimental/grpc annotations

* k6: bump k6 to v0.45

* Add browser options

---------

Co-authored-by: Oleg Bespalov <oleg.bespalov@grafana.com>

Commit:
🤖 Merge PR #65734 [react] react@next is now react@canary by @eps1lon

* [react] react@next is now react@canary

* Same for react-dom

* Same for react-is

samantharamon committed:
Remove nested bullet points (#65794)

Page- committed:
🤖 Merge PR #65824 [netmask]: Fix regressions in 2.0.0 typings by @Page-

* [netmask]: Re-add properties `maskLong` and `netLong`

* [netmask]: Mark functions as methods and not properties

* [netmask]: Add test that we can extend the Netmask class

AleFossati commented:
Fran, acho que o `format` deu uma dificultada novamente para identificarmos o que é alteração e o que é formatação. Poderia tirar os ajustes de formatação, pls?

Eu desativaria o `format on save` do vscode e faria o reset da branch pra três commits atrás, descartando os três do print. Assim daria pra fazê-los novamente sem formatação.


![image](https://github.com/francineemilia/DefinitelyTyped/assets/104934463/18b52ac3-c867-4928-8c57-7295d2f6117f)

ienzam committed:
🤖 Merge PR #65807 Jasmine: Remove Enzam from maintainers list. by @ienzam

mstoykov committed:
🤖 Merge PR #65798 Fix prettier command to not expand in shell by @mstoykov

* Fix prettier command to not expand in shell

The previous prettier command was expanded by the shell used which might differ from how it will be expanded by prettier.

Which effectively means it won't go through all subdirectories.

* Make the changes in the translated README files

JoshuaKGoldberg committed:
🤖 Merge PR #65815 Deprecate @types/node-emoji as of node-emoji@2.1.0 by @JoshuaKGoldberg

github-actions[bot] committed:
Remove contributors with deleted accounts (#65817)

* Remove contributors with deleted accounts #no-publishing-comment

* Update types/zwjsbridge/index.d.ts

Co-authored-by: Jungzl <35426360+Jungzl@users.noreply.github.com>

* Update types/zwlog-browser/index.d.ts

Co-authored-by: Jungzl <35426360+Jungzl@users.noreply.github.com>

---------

Co-authored-by: TypeScript Bot <typescriptbot@microsoft.com>
Co-authored-by: Gabriela Araujo Britto <gabrielaa@microsoft.com>
Co-authored-by: Jungzl <35426360+Jungzl@users.noreply.github.com>

joao-mbn committed:
🤖 Merge PR #65680 Added @types/hot-formula-parser by @joao-mbn

* Added @types/hot-formula-parser

* Added missing error exports.

* Added correct return for Parser setters. Removed built-in variables from export, added them as to description of setVariables method.

* Added dual mode. Patched CallCell hooks types.

* Removed type from package.json and changes .mts export syntax to that of esm.

yutak23 committed:
🤖 Merge PR #65820 feat: Add types for kebabcase-keys by @yutak23

francineemilia committed:
chore: changes in bricks settings

francineemilia committed:
chore: add tests to new settings

francineemilia committed:
chore: format archives

AleFossati committed:
🤖 Merge PR #65833 Added PSE types by @AleFossati

* added new form properties to pse and 3ds setup

* fix bricks style as optional

* fix MercadoPago second parameter as optional

* added entity type to initialization

* added semicolon

* formatting

AleFossati commented on types/mercadopago-sdk-js/modules/bricks.d.ts:
```diff
@@ -1,328 +1,335 @@
+        message: string;
+    }
+
+    interface iBrickCallbacks {
```
```suggestion
    interface BrickCallbacks {
```

AleFossati commented on types/mercadopago-sdk-js/modules/bricks.d.ts:
```diff
@@ -1,328 +1,335 @@
+        onError?: (error: BrickError) => void;
+    }
+
+    interface onSubmit<BrickType> {
```
podemos remover o `on` da interface

```suggestion
    interface Submit<BrickType> {
```

AleFossati commented on types/mercadopago-sdk-js/modules/bricks.d.ts:
```diff
@@ -1,328 +1,335 @@
+        ) => BrickType extends 'wallet' ? Promise<string> : Promise<void>;
+    }
+
+    interface onBinChange {
```
podemos remover o `on` da interface

```suggestion
    interface BinChange {
```

AleFossati commented:
Show Fran, ficou bem mais fácil, obrigado!
Só tenho algumas sugestões de nomenclatura pra gente tentar manter um padrão, geralmente tipos e interfaces a gente usa `PascalCase` e conversamos com core-cho recentemente pra não adicionarmos `I` e `T` antes dos nomes dos tipos e interfaces.

francineemilia committed:
Update types/mercadopago-sdk-js/modules/bricks.d.ts

Co-authored-by: Alexandre Fossati <alexandre.fossati@mercadolivre.com>

francineemilia committed:
Update types/mercadopago-sdk-js/modules/bricks.d.ts

Co-authored-by: Alexandre Fossati <alexandre.fossati@mercadolivre.com>

francineemilia committed:
Update types/mercadopago-sdk-js/modules/bricks.d.ts

Co-authored-by: Alexandre Fossati <alexandre.fossati@mercadolivre.com>

andreagostinho-meli commented on types/mercadopago-sdk-js/modules/bricks.d.ts:
```diff
@@ -1,328 +1,335 @@
+    interface Submit<BrickType> {
+        onSubmit: (
+            formData: CardFormData | PaymentFormData,
+            param2?: AdditionalCardFormData | AdditionalPaymentFormData,
```
Na documentação nós chamamos esse parâmetro de `additionalData`
```suggestion
            additionalData?: AdditionalCardFormData | AdditionalPaymentFormData,
```

andreagostinho-meli commented on types/mercadopago-sdk-js/modules/bricks.d.ts:
```diff
@@ -1,328 +1,335 @@
+        BUY = 'buy',
+    }
+
+    enum WalletButtonValueProp {
+        PRACTICALITY = 'practicality',
+        CONVENIENCE = 'convenience',
+        SECURITY_DETAILS = 'security_details',
+        SECURITY_SAFETY = 'security_safety',
+    }
```
Aqui estão faltando as novas value props, não?  `convenience_all`, `convenience_credits` e `smart_option`.
cc @meliguicarvalho para verificar.

francineemilia replied:
@andreagostinho-meli no caso das novas value props, elas estão no commit do Isma, esta PR é anterior ao release, daí a diferença de escopo do PR

samantharamon committed:
Create the SelectedItemDetails interface and add additional multi-select properties in preview (#65784)

mattbford committed:
🤖 Merge PR #65805 [@types/aws-param-store] Add type definitions for put methods by @mattbford

* [aws-param-store] Add type definitions for put methods

* Add tests

* Add legacy type and add put to query return types

* Add missing semicolon

francineemilia committed:
Update types/mercadopago-sdk-js/modules/bricks.d.ts

Co-authored-by: André <andre.agostinho@mercadolivre.com>

plbstl committed:
🤖 Merge PR #65730 [minimasonry] Initial types for minimasonry by @plbstl

* add types for minimasonry 1.3

* update minimasonry types to be used through a default ESM import

* update minimasonry types to allow use as a browser script

* remove dtslint errors

- strict-export-declare-modifiers
- npm-naming

ImRodry committed:
🤖 Merge PR #65838 chore: remove @aero/sanitizer types by @ImRodry

francineemilia committed:
Merge branch 'master' into fix/CHOBK-727
